lib/full_url_for.js

```diff
@@ -18,11 +18,14 @@ function fullUrlForHelper(path = '/') {
   return cache.apply(`${config.url}-${prettyUrlsOptions.trailing_index}-${prettyUrlsOptions.trailing_html}-${path}`, () => {
     if (/^(\/\/|http(s)?:)/.test(path)) return path;
 
-    const sitehost = parse(config.url).hostname || config.url;
-    const data = new URL(path, `http://${sitehost}`);
-
-    // Exit if input is an external link or a data url
-    if (data.hostname !== sitehost || data.origin === 'null') return path;
+    const config_url = config.url;
+    // if sitehost is empty, skip this step
+    if (config_url !== '' && config_url !== null) {
```

Comment:
It is not safe, in case no `config.url` is given the `undefined` will be passed then.

Reply:
@SukkaW  You are right, and I add code and test cases for handling the null and undefined conditions.